src/main/java/com/direwolf20/buildinggadgets/common/network/PacketToggleMode.java

```diff
@@ -44,21 +44,21 @@ public IMessage onMessage(PacketToggleMode message, MessageContext ctx) {
         private void handle(PacketToggleMode message, MessageContext ctx) {
             EntityPlayerMP playerEntity = ctx.getServerHandler().player;
             ItemStack heldItem = playerEntity.getHeldItem(EnumHand.MAIN_HAND);
-            if (!(heldItem.getItem() instanceof GenericGadget)) {
+            if (!(heldItem.getItem() instanceof GadgetGeneric)) {
                 heldItem = playerEntity.getHeldItemOffhand();
-                if (!(heldItem.getItem() instanceof GenericGadget)) {
+                if (!(heldItem.getItem() instanceof GadgetGeneric)) {
                     return;
                 }
             }
-            if (!heldItem.isEmpty() && heldItem.getItem() instanceof BuildingTool) {
-                BuildingTool buildingTool = (BuildingTool) (heldItem.getItem());
-                buildingTool.setMode(playerEntity, heldItem, message.mode);
-            } else if (!heldItem.isEmpty() && heldItem.getItem() instanceof ExchangerTool) {
-                ExchangerTool exchangerTool = (ExchangerTool) (heldItem.getItem());
-                exchangerTool.setMode(playerEntity, heldItem, message.mode);
-            } else if (!heldItem.isEmpty() && heldItem.getItem() instanceof CopyPasteTool) {
-                CopyPasteTool copyPasteTool = (CopyPasteTool) (heldItem.getItem());
-                copyPasteTool.setMode(playerEntity, heldItem, message.mode);
+            if (!heldItem.isEmpty() && heldItem.getItem() instanceof GadgetBuilding) {
```

Comment:
Maybe move this heldItem.isEmpty check to an enclosing if, as to avoid verbose multiple isEmpty checks?